src/rt/minfo.d

```diff
@@ -457,10 +457,15 @@ struct ModuleGroup
 
             // add this module to the construction order list
             ctors[ctoridx++] = current;
+            return true;
         }
 
-        immutable(ModuleInfo)*[] doSort(size_t relevantFlags)
+        // `cycle` is set to `true` if deprecated cycle error otherwise set `result`.
+        void doSort(size_t relevantFlags, ref bool cycle, ref immutable(ModuleInfo)*[] result)
```

Comment:
Hm... what about returning `bool`, and then if statement below reads:

```D
if (doSort(...) || doSort(...))
```

In any case, the new code is correct at least.